test/runnable/test18322.d

```diff
@@ -0,0 +1,21 @@
+/*
+REQUIRED_ARGS: -Irunnable/imports
+COMPILED_IMPORTS: imports/test18322import.d
+PERMUTE_ARGS:
+*/
+import test18322import;
+void main(){
+    version(Windows)
+        auto sep = "\\";
+    else
+        auto sep = "/";
+
+    auto filename = "runnable" ~ sep ~ "test18322.d";
+
+    fun(filename);
+    mixin(`fun(filename ~ "-mixin-16");`);
```

Comment:
That was unexpected for me. I didn't expected that `__FILE_FULL_PATH__` changes for mixins to sth. like `test18322.d-mixin-16`, but it's really the current behavior - https://run.dlang.io/is/F4ixWD

Reply:
Hmm, yeah this is an interesting case.  I would expect `__FILE__` to match `__FILE_FULL_PATH__` though, and `__FILE__` goes to the mixin.  I think if we want to change it we should probably change both...